chalk-parse/src/parser.lalrpop

```diff
@@ -130,6 +131,17 @@ ProjectionTy: ProjectionTy = {
     },
 };
 
+UnselectedProjectionTy: UnselectedProjectionTy = {
+    "(" <ty:Ty> "::" <name:Id> <a:Angle<Parameter>> ")" => {
```

Comment:
Left recursion itself should not be a problem -- do you mean that it reports an ambiguity or other error to you?

Reply:
Without the parentheses LALRPOP crashes with:
```
thread 'main' panicked at 'index out of bounds: the len is 0 but the index is 0', /home/user/.cargo/registry/src/github.com-1ecc6299db9ec823/lalrpop-0.13.1/src/lr1/error/mod.rs:545:33
```
Somehow I thought left recursion and LR parsers didn't like each other... (Apparently, this isn't true at all.) I don't know what the issue is. I have noticed before that LALRPOP crashes when trying to report certain conflicts. (I should have opened a bug report.)

Reply:
Found it! `Ty` already has a right recursive rule (`"for" "<" Comma<LifetimeId> ">" Ty`). I'll give the `for` rule a lower precedence.